stock_account_internal_move/models/stock_move.py

```diff
@@ -0,0 +1,66 @@
+# Copyright (C) 2018 by Camptocamp
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+from odoo import api, models
+from odoo.tools import float_round
+
+
+class StockMove(models.Model):
+    _inherit = 'stock.move'
+
+    @api.multi
+    def _action_done(self):
+        res = super()._action_done()
+
+        for move in res:
+            # first of all, define if we need to even valuate something
+            if move.product_id.valuation != 'real_time':
+                continue
+            # we're customizing behavior on moves between internal locations
+            # only, thus ensuring that we don't clash w/ account moves
+            # created in `stock_account`
+            if not (move.location_id.usage
+                    == move.location_dest_id.usage
+                    == 'internal'):
+                continue
+
+            # treated by `super()` as a move w/ negative qty due to this hunk:
+            # quantity = self.product_qty or context.get('forced_quantity')
+            # quantity = quantity if self._is_in() else -quantity
+            # so, move qty is flipped twice and thus preserved
+            move = move.with_context(forced_quantity=-move.product_qty)
+
+            location_from = move.location_id
+            location_to = move.location_dest_id
+
+            # get valuation accounts for product
+            # done in _get_accounting_data_for_valuation?
+            product_valuation_accounts \
+                = move.product_id.product_tmpl_id.get_product_accounts()
+            stock_valuation = product_valuation_accounts.get('stock_valuation')
+            stock_journal = product_valuation_accounts.get('stock_journal')
+
+            # calculate move cost
+            # TODO: recheck if this part respects product valuation method
+            move.value = float_round(
+                value=move.product_id.standard_price * move.quantity_done,
+                precision_rounding=move.company_id.currency_id.rounding,
+            )
```

Comment:
@arkostyuk I suggest to put the computation of the cost in a separate method that we can extend in another module.